api/controllers/v2/github.ts

```diff
@@ -1,71 +1,125 @@
-    email: githubEvent.pusher.email ?? '',
+    title: title,
+    user: githubEvent.sender.login,
     status: JobStatus.inQueue,
     createdTime: new Date(),
     startTime: null,
     endTime: null,
     priority: 1,
     error: {},
     result: null,
-    payload: {
-      jobType: 'githubPush',
-      source: 'github',
-      action: 'push',
-      repoName: githubEvent.repository.name,
-      branchName: githubEvent.ref.split('/')[2],
-      isFork: githubEvent.repository.fork,
-      repoOwner: githubEvent.repository.owner.login,
-      url: githubEvent.repository.clone_url,
-      newHead: githubEvent.after,
-      urlSlug: urlSlug,
-      prefix: prefix,
-      project: project,
-      directory: directory,
-    },
+    payload: payload,
     logs: [],
   };
 }
 
-export const TriggerBuild = async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
+interface CreatePayloadProps {
+  repoName: string;
+  isSmokeTestDeploy?: boolean;
+  prefix: string;
+  repoBranchesRepository: RepoBranchesRepository;
+  repoInfo: ReposBranchesDocsetsDocument;
+  newHead?: string;
+  repoOwner?: string;
+  directory?: string;
+  githubEvent?: PushEvent;
+}
+
+async function createPayload({
+  repoName,
+  isSmokeTestDeploy = false,
+  prefix,
+  repoBranchesRepository,
+  repoInfo,
+  newHead,
+  repoOwner = '',
+  githubEvent,
+  directory,
+}: CreatePayloadProps): Promise<EnhancedPayload> {
+  const source = 'github';
+  const project = repoInfo?.project ?? repoName;
+
+  let branchName: string;
+  let jobType: string;
+  let action: string;
+  let url: string;
+
+  if (isSmokeTestDeploy) {
+    url = 'https://github.com/' + repoOwner + '/' + repoName;
+    action = 'automatedTest';
+    jobType = 'productionDeploy';
+    branchName = 'master';
+  } else {
+    if (!githubEvent) {
+      throw 'Non SmokeTest Deploy jobs must have a github Event';
+    }
+    action = 'push';
+    jobType = 'githubPush';
+    branchName = githubEvent.ref.split('/')[2];
+    url = githubEvent?.repository.clone_url;
+    newHead = githubEvent?.after;
+    repoOwner = githubEvent.repository.owner.login;
+  }
+
+  const branchInfo = await repoBranchesRepository.getRepoBranchAliases(repoName, branchName, repoInfo.project);
+  const urlSlug = branchInfo.aliasObject?.urlSlug ?? branchName;
+
+  return {
+    jobType,
+    source,
+    action,
+    repoName,
+    repoOwner,
+    branchName,
+    project,
+    prefix,
+    urlSlug,
+    url,
+    newHead,
+    directory,
+  };
+}
+
+export const triggerSmokeTestAutomatedBuild = async (event: APIGatewayEvent): Promise<APIGatewayProxyResult | null> => {
   const client = new mongodb.MongoClient(c.get('dbUrl'));
   await client.connect();
   const db = client.db(c.get('dbName'));
   const consoleLogger = new ConsoleLogger();
   const jobRepository = new JobRepository(db, c, consoleLogger);
+  const projectsRepository = new ProjectsRepository(client.db('docs_metadata'), c, consoleLogger);
```

Comment:
I think it would be good to put the DB name `'docs_metadata'` as a const variable in case that we use this repository somewhere else. This would make sure we don't accidentally misspell the name of the database somewhere else.